charts/backstage/templates/backstage-deployment.yaml

```diff
@@ -68,6 +73,10 @@ spec:
             - {{ .filename | quote }}
           {{- end }}
           {{- end }}
+          {{- if .Values.backstage.appConfig }}
+            - "--config"
+            - "app-config-from-configmap.yaml"
+          {{- end }}
```

Comment:
Open question: If we put this after config args from `Values.backstage.extraAppConfig` the `.Values.backstage.appConfig` will have [higher priority](https://backstage.io/docs/conf/writing#configuration-files). Is this desired behavior or do we want the opposite?

cc @backstage/helm-charts

Reply:
The truth is that, with the current approach, appConfig is just another extraAppConfig, it just generates the ConfigMap without the need for other interaction. So even the name "appConfig" does not really sound right.

I do not really know about the priority..both this appConfig and the ConfigMap created manually would be done at a very late stage of a deployment (vs the configs included in the OCI image built previously). I do not see any argument for one of the other to have higher priority, but for sure I need to clearly state and document which one has it, so good point @tumido! :)

Reply:
@froblesmartin well.. the priority is defined by (besides other things) the order of `--config` arguments, see the link posted above:

![image](https://user-images.githubusercontent.com/7453394/207820630-ad015641-5c97-42a0-bf07-ce7d764a9c2a.png)

What I'm trying to say is that: Does it make more sense to have:

1. `appConfig` from your PR to potentially override values found in `extraAppConfig`s (that's how it's currently implemented here)

Or

2. `appConfig` to be overriden by whatever appears in any of the `extraAppConfig`s

In either case this behavior is not described to the end user anywhere and we should make it more straightforward in that regard to avoid user confusion aka "why the heck is not my config being applied when I clearly state it here?".

Reply:
I'm a little lost at the need for the extraAppConfig piece,

What is the purpose of `appConfig` within the values.yaml and templates  if not for adding appconfigs with the `extraVolumes`  and   `extraVolumeMounts` ?

Also can I suggest we don't use the `common.tplvalues.render` this forces a dependency on the bitnami common charts that isn't necessary for such simple templating ( especially for users who are using an external postgres DB)

Reply:
`bitnami/common` is already a dependency and we use it in various other places across this chart already. I don't see it as a problematic overhead. Helm won't render this chart without bitnami common regardless of this PR.

I think having the backstage config inlined directly from the `values.yaml` greatly simplifies the install process since you don't have to install the helm chart and also create a separate configmap, you can do both in the same step. The biggest advantage here is that you can dynamically template things in here, further reducing copy pasting and making the chart more adaptable:

Imagine following values file:
```yaml
backstage:
  appConfig:
    app:
      baseUrl: https://{{ .Values.ingress.host}}
ingress:
  enabled: true
  host: foobar.baz
```

Now you have the `host` variable defined only once in the whole config and you don't have to copypaste it across your configmaps or into multiple places of the values file. I personally think that's desirable.

Reply:
@tumido following with the priority topic, as I said in my previous comment, I do not really know which should have more priority, but if we agree on which one should, then I can document it ;)

For now, just because it is the current way, I would leave it like that, having higher priority what is defined in appConfig.

Reply:
Ok, let's leave it as is :slightly_smiling_face: